src/js/jsx/sections/layers/LayerFace.jsx

```diff
@@ -65,53 +76,34 @@ define(function (require, exports, module) {
             };
         },
         
+        componentWillMount: function () {
+            this.getFlux().stores.document.addLayerFaceListener(this.props.documentID, this.props.layerID,
+                this._handleLayerFaceChange);
+        },
+        
+        componentWillUnmount: function () {
+            this.getFlux().stores.document.removeLayerFaceListener(this.props.documentID, this.props.layerID,
+                this._handleLayerFaceChange);
+        },
+
         shouldComponentUpdate: function (nextProps, nextState) {
-            // Drag states
+            var depth = nextProps.depth,
+                isLayerChanged = !!nextProps.changedLayerPaths[depth] &&
+                    nextProps.changedLayerPaths[depth].has(nextProps.layerID);
+
+            if (isLayerChanged) {
```

Comment:
This part respond to `layer-tree-change` event initiated by root LayerList. It checks whether its layer is within the changed layer path (for example, when the layer's children is deleted / create / re-ordered). Otherwise it will skip its rendering (and its children)